platform/osx/os_osx.mm

```diff
@@ -1167,30 +1165,55 @@ static void displays_arrangement_changed(CGDirectDisplayID display_id, CGDisplay
 	if (cursor_shape == p_shape)
 		return;
 
-	switch (p_shape) {
-		case CURSOR_ARROW: [[NSCursor arrowCursor] set]; break;
-		case CURSOR_IBEAM: [[NSCursor IBeamCursor] set]; break;
-		case CURSOR_POINTING_HAND: [[NSCursor pointingHandCursor] set]; break;
-		case CURSOR_CROSS: [[NSCursor crosshairCursor] set]; break;
-		case CURSOR_WAIT: [[NSCursor arrowCursor] set]; break;
-		case CURSOR_BUSY: [[NSCursor arrowCursor] set]; break;
-		case CURSOR_DRAG: [[NSCursor closedHandCursor] set]; break;
-		case CURSOR_CAN_DROP: [[NSCursor openHandCursor] set]; break;
-		case CURSOR_FORBIDDEN: [[NSCursor arrowCursor] set]; break;
-		case CURSOR_VSIZE: [[NSCursor resizeUpDownCursor] set]; break;
-		case CURSOR_HSIZE: [[NSCursor resizeLeftRightCursor] set]; break;
-		case CURSOR_BDIAGSIZE: [[NSCursor arrowCursor] set]; break;
-		case CURSOR_FDIAGSIZE: [[NSCursor arrowCursor] set]; break;
-		case CURSOR_MOVE: [[NSCursor arrowCursor] set]; break;
-		case CURSOR_VSPLIT: [[NSCursor resizeUpDownCursor] set]; break;
-		case CURSOR_HSPLIT: [[NSCursor resizeLeftRightCursor] set]; break;
-		case CURSOR_HELP: [[NSCursor arrowCursor] set]; break;
-		default: {};
+	if (cursors[p_shape] != NULL) {
+		[cursors[p_shape] set];
+	} else {
+		switch (p_shape) {
+			case CURSOR_ARROW: [[NSCursor arrowCursor] set]; break;
+			case CURSOR_IBEAM: [[NSCursor IBeamCursor] set]; break;
+			case CURSOR_POINTING_HAND: [[NSCursor pointingHandCursor] set]; break;
+			case CURSOR_CROSS: [[NSCursor crosshairCursor] set]; break;
+			case CURSOR_WAIT: [[NSCursor arrowCursor] set]; break;
+			case CURSOR_BUSY: [[NSCursor arrowCursor] set]; break;
+			case CURSOR_DRAG: [[NSCursor closedHandCursor] set]; break;
+			case CURSOR_CAN_DROP: [[NSCursor openHandCursor] set]; break;
+			case CURSOR_FORBIDDEN: [[NSCursor arrowCursor] set]; break;
+			case CURSOR_VSIZE: [[NSCursor resizeUpDownCursor] set]; break;
+			case CURSOR_HSIZE: [[NSCursor resizeLeftRightCursor] set]; break;
+			case CURSOR_BDIAGSIZE: [[NSCursor arrowCursor] set]; break;
+			case CURSOR_FDIAGSIZE: [[NSCursor arrowCursor] set]; break;
+			case CURSOR_MOVE: [[NSCursor arrowCursor] set]; break;
+			case CURSOR_VSPLIT: [[NSCursor resizeUpDownCursor] set]; break;
+			case CURSOR_HSPLIT: [[NSCursor resizeLeftRightCursor] set]; break;
+			case CURSOR_HELP: [[NSCursor arrowCursor] set]; break;
+			default: {};
+		}
 	}
 
 	cursor_shape = p_shape;
 }
 
+void OS_OSX::set_custom_mouse_cursor(const RES &p_cursor, CursorShape p_shape, const Vector2 &p_hotspot){
```

Comment:
Exported project will not work, I updated this method in this commit 83f073a00b31f6aaafe90ec80cfdd61c9982cc6b, can you update please?

Reply:
Thank you for this catch. I updated the merge request

Reply:
With your fix, it does not compile anymore for osx

Reply:
Yes, I forgot to say about these two lines, now I think it's right.

Reply:
`PoolVector` structure does not exist in Godot 2.1.4. I'm trying to figure out what I should use instead.

Reply:
oh wait, change ```image->``` to ```image.``` and
```
// from
int len = image->get_width() * image->get_height();
PoolVector<uint8_t> data = image->get_data();
PoolVector<uint8_t>::Read r = data.read();

/* Premultiply the alpha channel */
for (int i = 0; i < len; i++) {
	uint8_t alpha = r[i * 4 + 3];
	pixels[i * 4 + 0] = (uint8_t)(((uint16_t)r[i * 4 + 0] * alpha) / 255);
	pixels[i * 4 + 1] = (uint8_t)(((uint16_t)r[i * 4 + 1] * alpha) / 255);
	pixels[i * 4 + 2] = (uint8_t)(((uint16_t)r[i * 4 + 2] * alpha) / 255);
	pixels[i * 4 + 3] = alpha;
}

// to
int len = img.get_width()*img.get_height();                       
DVector<uint8_t> data = img.get_data();                           
DVector<uint8_t>::Read r = data.read();                           
                                                                  
/* Premultiply the alpha channel */                               
for (int i = 0; i<len ; i++) {                                    
  uint8_t alpha = r[i*4+3];                                       
  pixels[i*4+0] = (uint8_t)(((uint16_t)r[i*4+0] * alpha) / 255);  
  pixels[i*4+1] = (uint8_t)(((uint16_t)r[i*4+1] * alpha) / 255);  
  pixels[i*4+2] = (uint8_t)(((uint16_t)r[i*4+2] * alpha) / 255);  
  pixels[i*4+3] = alpha;                                          
                                                                  
}                                                                 
```

Reply:
Thank you, already done :-) Its compiling for OSX

Reply:
Nice 👍

Reply:
Thank you for your amazing help for OSX 💯